File: backend/src/main/java/ma/zs/alc/service/facade/admin/course/SectionItemAdminService.java
package ma.zs.alc.service.facade.admin.course;

import java.util.List;
import ma.zs.alc.bean.core.course.SectionItem;
import ma.zs.alc.dao.criteria.core.course.SectionItemCriteria;
import ma.zs.alc.zynerator.service.IService;



public interface SectionItemAdminService extends  IService<SectionItem,SectionItemCriteria>  {

    List<SectionItem> findBySectionId(Long id);
    int deleteBySectionId(Long id);
    long countBySectionCode(String code);




}

File: frontend-bqckup/src/app/shared/service/admin/course/SectionItemAdmin.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';

import {environment} from '../../../../../environments/environment';

import {SectionItemDto} from '../../../model/course/SectionItem.model';
import {SectionItemCriteria} from '../../../criteria/course/SectionItemCriteria.model';
import {AbstractService} from "../../../../zynerator/service/AbstractService";


@Injectable({
    providedIn: 'root'
})
export class SectionItemAdminService extends AbstractService<SectionItemDto, SectionItemCriteria> {
    constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    get API() {
        return environment.apiUrlAlcservice + 'admin/sectionItem/';
    }

    public constrcutDto(): SectionItemDto {
        return new SectionItemDto();
    }

    public constrcutCriteria(): SectionItemCriteria {
        return new SectionItemCriteria();
    }
}
